### PR TITLE
CORTX-28915: Modify HA code to use changes from cortx-utils for Health Event creation

### PR DESCRIPTION
**Pull request by gauri-bhosale**
# Problem Statement
- Modify HA code to use changes from cortx-utils for Health Event creation
- https://jts.seagate.com/browse/CORTX-28915

# Design
- Move Event generic class and Health event class to cortx-utils.
- Alter health event creation and handling according to cortx-utils in HA code

# Coding
-  [X] Coding conventions are followed and code is consistent

# Testing 
- [X] Unit and System Tests are added
- [X] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [X] Testing was performed with RPM

# Review Checklist 
- [X] PR is self reviewed
- [X] JIRA number/GitHub Issue added to PR
- [X] Jira and state/status is updated and JIRA is updated with PR link
- [X] Check if the description is clear and explained
- [] Is there a change in filename/package/module or signature? [Y/N]: N 
- [] If yes for above point, is a notification sent to all other cortx components? [Y/N]: N
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]N
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Documentation
  Checklist for Author
- [X] Changes done to WIKI / Confluence page / Quick Start Guide

Log Result : 
Fault_tolerance.log

2022-03-08 03:37:00 fault_tolerance [8]: INFO [filter_event] This alert needs an attention: resource_type: node
2022-03-08 03:37:00 fault_tolerance [8]: INFO [process_event] SystemHealth: Processing node:node:89e0d99c15a549ddbdffbe018e2ab928 with status online
2022-03-08 03:37:00 fault_tolerance [8]: INFO [process_message] Received the message from message bus: b'{"header": {"version": "1.0", "timestamp": "1646710620.26266", "event_id": "1646710620.26266cf04cbd032a8486097409128510a836c"}, "payload": {"source": "", "cluster_id": "", "site_id": "", "rack_id": "", "storageset_id": "", "node_id": "97773057d6444bbebf731af3b017d1d6", "resource_type": "node", "resource_id": "97773057d6444bbebf731af3b017d1d6", "resource_status": "online", "specific_info": {"generation_id": "cortx-server-ssc-vm-rhev4-2221-8486d9f4dc-gp2n5", "is_status": "True"}}}'

2022-03-07 12:17:40 fault_tolerance [8]: INFO [process_event] SystemHealth: Processing node:node:89e0d99c15a549ddbdffbe018e2ab928 with status online
2022-03-07 12:17:40 fault_tolerance [8]: INFO [process_message] Received the message from message bus: b'{"header": {"version": "1.0", "timestamp": "1646655460.5213675", "event_id": "1646655460.52136755de50fa0f1ed4bb9bdaff352b43c59e5"}, "payload": {"source": "", "cluster_id": "", "site_id": "", "rack_id": "", "storageset_id": "", "node_id": "97773057d6444bbebf731af3b017d1d6", "resource_type": "node", "resource_id": "97773057d6444bbebf731af3b017d1d6", "resource_status": "online", "specific_info": {"generation_id": "cortx-server-ssc-vm-rhev4-2221-8486d9f4dc-gp2n5"}}}'
2022-03-07 12:17:40 fault_tolerance [8]: INFO [init_evaluators] Initialize all the health evaluator elements
2022-03-07 12:17:40 fault_tolerance [8]: INFO [init_evaluators] HealthEvaluator ha.core.system_health.health_evaluators.cluster_health_evaluator.ClusterHealthEvaluator is initalized...
2022-03-07 12:17:40 fault_tolerance [8]: INFO [init_evaluators] HealthEvaluator ha.core.system_health.health_evaluators.site_health_evaluator.SiteHealthEvaluator is initalized...

k8s_monitor.log : 
2022-03-07 12:18:35 k8s_resource_monitor [7]: INFO [publish_alert] pod_monitor sending alert on message bus {"header": {"version": "1.0", "timestamp": "1646655515.607267", "event_id": "1646655515.6072679368b55131cb4423b184597381d44b75"}, "payload": {"source": "", "cluster_id": "", "site_id": "", "rack_id": "", "storageset_id": "", "node_id": "97773057d6444bbebf731af3b017d1d6", "resource_type": "node", "resource_id": "97773057d6444bbebf731af3b017d1d6", "resource_status": "online", "specific_info": {"generation_id": "cortx-server-ssc-vm-rhev4-2221-8486d9f4dc-gp2n5"}}}

health_monitor.log 
 INFO [evaluate] Evaluated action ['publish'] for key action/node/online
2022-03-08 07:29:18 health_monitor [7]: INFO [process_event] Evaluated {"source": "", "event_id": "1646724555.937822626f326bf03ec41c7a60234b0a4ed46a9", "event_type": "online", "severity": "informational", "site_id": "0xBAADFOOD", "rack_id": "0xBAADFOOD", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "storageset_id": "97773057d6444bbebf731af3b017d1d6", "node_id": "97773057d6444bbebf731af3b017d1d6", "host_id": "97773057d6444bbebf731af3b017d1d6", "resource_type": "node", "timestamp": "1646724555.9378226", "resource_id": "97773057d6444bbebf731af3b017d1d6", "specific_info": {"generation_id": "cortx-server-ssc-vm-rhev4-2221-8486d9f4dc-gp2n5", "is_status": "True"}} with action ['publish']
2022-03-08 07:29:18 health_monitor [7]: INFO [act] Default action handler with Event: 1646724555.937822626f326bf03ec41c7a60234b0a4ed46a9 and actions : ['publish']
2022-03-08 07:29:18 health_monitor [7]: INFO [publish] Sending action event {"header": {"version": "1.0", "timestamp": "1646724558.199793", "event_id": "1646724558.1997939d710554c1c54a14ba96195d5e330e81"}, "payload": {"source": "", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "site_id": "0xBAADFOOD", "rack_id": "0xBAADFOOD", "storageset_id": "97773057d6444bbebf731af3b017d1d6", "node_id": "97773057d6444bbebf731af3b017d1d6", "resource_type": "node", "resource_id": "97773057d6444bbebf731af3b017d1d6", "resource_status": "online", "specific_info": {"generation_id": "cortx-server-ssc-vm-rhev4-2221-8486d9f4dc-gp2n5", "is_status": "True"}}} to component hare
2022-03-08 07:29:18 health_monitor [7]: INFO [evaluate] Evaluated action [] for key action/rack/fault_resolved
2022-03-08 07:29:18 health_monitor [7]: INFO [evaluate] Evaluate

Node failed : 
2022-03-08 07:33:12 health_monitor [7]: INFO [evaluate] Evaluated action ['publish'] for key action/node/failed
2022-03-08 07:33:12 health_monitor [7]: INFO [process_event] Evaluated {"source": "", "event_id": "1646724792.254001e8b2f99b1bed4ad28089825a5e2838e0", "event_type": "failed", "severity": "error", "site_id": "0xBAADFOOD", "rack_id": "0xBAADFOOD", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "storageset_id": "9b3dbeb08a444dc1a98c14025c4776d2", "node_id": "9b3dbeb08a444dc1a98c14025c4776d2", "host_id": "9b3dbeb08a444dc1a98c14025c4776d2", "resource_type": "node", "timestamp": "1646724792.254001", "resource_id": "9b3dbeb08a444dc1a98c14025c4776d2", "specific_info": {"generation_id": "cortx-data-ssc-vm-rhev4-2222-7d55d95c47-84lvr"}} with action ['publish']
2022-03-08 07:33:12 health_monitor [7]: INFO [act] Default action handler with Event: 1646724792.254001e8b2f99b1bed4ad28089825a5e2838e0 and actions : ['publish']
2022-03-08 07:33:12 health_monitor [7]: INFO [publish] Sending action event {"header": {"version": "1.0", "timestamp": "1646724792.5692775", "event_id": "1646724792.569277513d14148928648f4a8a32d616ce9e39b"}, "payload": {"source": "", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "site_id": "0xBAADFOOD", "rack_id": "0xBAADFOOD", "storageset_id": "9b3dbeb08a444dc1a98c14025c4776d2", "node_id": "9b3dbeb08a444dc1a98c14025c4776d2", "resource_type": "node", "resource_id": "9b3dbeb08a444dc1a98c14025c4776d2", "resource_status": "failed", "specific_info": {"generation_id": "cortx-data-ssc-vm-rhev4-2222-7d55d95c47-84lvr"}}} to component hare
2022-03-08 07:33:13 health_monitor [7]: INFO [evaluate] Evaluated action [] for key action/rack/threshold_breached:low

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: ha/monitor/k8s/parser.py [hunk 1]
```diff
@@ -24,18 +24,18 @@
...
 
 class ObjectParser:
     def __init__(self):
-        self.payload = {HEALTH_ATTRIBUTES.SOURCE.value: HEALTH_EVENT_SOURCES.MONITOR.value, \
-                    HEALTH_ATTRIBUTES.CLUSTER_ID.value: None, HEALTH_ATTRIBUTES.SITE_ID.value: None, \
-                    HEALTH_ATTRIBUTES.RACK_ID.value: None, HEALTH_ATTRIBUTES.STORAGESET_ID.value: None, \
-                    HEALTH_ATTRIBUTES.NODE_ID.value: None, HEALTH_ATTRIBUTES.RESOURCE_TYPE.value: None, \
-                    HEALTH_ATTRIBUTES.RESOURCE_ID.value: None, HEALTH_ATTRIBUTES.RESOURCE_STATUS.value: None, \
-                    HEALTH_ATTRIBUTES.SPECIFIC_INFO.value: {}}
+        self.payload = {HealthAttr.SOURCE: HEALTH_EVENT_SOURCES.MONITOR.value, \
```
Comment (mariyappanp) [inline ha/monitor/k8s/parser.py:33]: "\" is not required in iterable object.

File: ha/core/event_manager/model/action_event.py [hunk 2]
```diff
@@ -16,8 +16,7 @@
 # cortx-questions@seagate.com.
 
 from ha.core.system_health.model.health_event import HealthEvent
-from ha.fault_tolerance.event import Event
-from ha.fault_tolerance.const import HEALTH_ATTRIBUTES
+from cortx.utils.event_framework.health import HealthAttr, HealthEvent as HEvent
```
Comment (mariyappanp) [inline ha/core/event_manager/model/action_event.py:19]: Why relative name used on HealthEvent?

**Comment by mariyappanp** [comment]
ha/fault_tolerance/event.py is removed.
No impact on HA test dir/unittest ? @gauri-bhosale

**Comment by ArchanaLimaye** [comment]
Please share testing details

**Comment by ArchanaLimaye** [comment]
ha/core/event_analyzer/parser/parser.py has conflicts with the base branch that need to be resolved

File: ha/monitor/k8s/parser.py (hunk 1 above)
Comment (gauri-bhosale) [reply ha/monitor/k8s/parser.py]: fixed.

**Comment by indrajitzagade** [comment]
@gauri-bhosale Please add the testing details to this PR.

File: ha/core/event_manager/event_manager.py [hunk 3]
```diff
@@ -384,26 +384,26 @@ def get_events(self, component : SUBSCRIPTION_LIST) -> list:
                     break
         return value
 
-    def publish(self, event: Event) -> None:
+    def publish(self, event: HealthEvent) -> None:
```
Comment (ArchanaLimaye) [inline ha/core/event_manager/event_manager.py:387]: class has changed from Event to HealthEvent 
Please confirm that this change has not broken any existing  functionality

**Comment by ArchanaLimaye** [comment]
Please confirm in testing that event from both monitor and Hare is getting created with all expected fields correctly populated after this change

File: ha/core/event_manager/model/action_event.py [hunk 4]
```diff
@@ -29,20 +28,19 @@ def __init__(self, healthevent: HealthEvent):
...
+            HealthAttr.RACK_ID: healthevent.rack_id,
+            HealthAttr.STORAGESET_ID: healthevent.storageset_id,
+            HealthAttr.NODE_ID: healthevent.node_id,
+            HealthAttr.RESOURCE_TYPE: healthevent.resource_type,
+            HealthAttr.RESOURCE_ID: healthevent.resource_id,
+            HealthAttr.RESOURCE_STATUS: healthevent.event_type,
         }
-        self.event.set_payload(payload)
+        self.event = HEvent(**payload)
+        self.event.set_specific_info({HealthAttr.SPECIFIC_INFO: healthevent.specific_info})
```
Comment (indrajitzagade) [inline ha/core/event_manager/model/action_event.py:43]: Do we need this key as well, HealthAttr.SPECIFIC_INFO?

Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py:43]: No as set_specific_info function handle this : 

    def set_payload_attr(self, key, val):
        super().set(f'payload>{key}', val)

def set_specific_info(self, spec_info: dict):
        """Set payload attribute of key/value pairs specific to the resource type."""
        specific_info = KvPayload(spec_info)
        for key in specific_info.get_keys():
            super().set_payload_attr(f'specific_info>{key}', specific_info.get(key))

File: ha/monitor/k8s/parser.py [hunk 5]
```diff
@@ -139,13 +139,13 @@ def _create_health_alert(self, res_type, res_name, health_status, generation_id)
...
         generation_id: name of the node in case of node alert
         """
-        self.event = Event()
-        self.payload[HEALTH_ATTRIBUTES.RESOURCE_TYPE.value] = res_type
-        self.payload[HEALTH_ATTRIBUTES.RESOURCE_ID.value] = self.payload[HEALTH_ATTRIBUTES.NODE_ID.value] = res_name
-        self.payload[HEALTH_ATTRIBUTES.RESOURCE_STATUS.value] = health_status
-        self.payload[HEALTH_ATTRIBUTES.SPECIFIC_INFO.value]["generation_id"] = generation_id
-        self.event.set_payload(self.payload)
-        return self.event.ret_dict()
+        self.event = HealthEvent()
```
[the remark this answers is not in the document]
Comment (gauri-bhosale) [reply ha/monitor/k8s/parser.py:142]: As in cortx-utils Event class is generic one, and Health Event class is specific to health HA events which is inherited y Event class.So we are using HealthEvent class in HA

File: ha/core/event_manager/event_manager.py (hunk 3 above)
Comment (gauri-bhosale) [reply ha/core/event_manager/event_manager.py:387]: In cortx-utils Event class is generic , HealthEvent is specific to HA ,where it inherited by Event class but has attributes of HA health realted.

File: ha/core/event_manager/model/action_event.py (hunk 4 above)
Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py:43]: in Utils it handle that : 
 def set_payload(self, payload: KvPayload):
        for key in payload.get_keys():
            super().set(f'payload>{key}', payload.get(key))


    def set_specific_info(self, spec_info: dict):
        """Set payload attribute of key/value pairs specific to the resource type."""
        specific_info = KvPayload(spec_info)
        for key in specific_info.get_keys():
            super().set_payload_attr(f'specific_info>{key}', specific_info.get(key))

**Comment by gauri-bhosale** [comment]
> @gauri-bhosale Please add the testing details to this PR.

added logs details.

**Review by mukhtar-inamdar — APPROVED** [review]

File: ha/core/event_manager/model/action_event.py [hunk 6]
```diff
@@ -29,20 +28,19 @@ def __init__(self, healthevent: HealthEvent):
...
+            HealthAttr.SITE_ID.value: healthevent.site_id,
+            HealthAttr.RACK_ID.value: healthevent.rack_id,
+            HealthAttr.STORAGESET_ID.value: healthevent.storageset_id,
+            HealthAttr.NODE_ID.value: healthevent.node_id,
+            HealthAttr.RESOURCE_TYPE.value: healthevent.resource_type,
+            HealthAttr.RESOURCE_ID.value: healthevent.resource_id,
+            HealthAttr.RESOURCE_STATUS.value: healthevent.event_type,
         }
-        self.event.set_payload(payload)
+        self.event = HEvent(**payload)
```
Comment (Madhura-08) [inline ha/core/event_manager/model/action_event.py:42]: Is it Hevent or HEvent? Please check.

Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py:42]: @Madhura-08  from cortx.utils.event_framework.health import HealthAttr, HealthEvent as HEvent line no 19 , there are 2 imports of HealthEvent, alias name

Comment (Madhura-08) [reply ha/core/event_manager/model/action_event.py:42]: Ok..

File: ha/core/event_manager/model/action_event.py [hunk 7]
```diff
@@ -29,20 +28,19 @@ def __init__(self, healthevent: HealthEvent):
...
-            HEALTH_ATTRIBUTES.SPECIFIC_INFO.value : healthevent.specific_info,
+            HealthAttr.SOURCE.value: healthevent.source,
+            HealthAttr.CLUSTER_ID.value: healthevent.cluster_id,
+            HealthAttr.SITE_ID.value: healthevent.site_id,
+            HealthAttr.RACK_ID.value: healthevent.rack_id,
+            HealthAttr.STORAGESET_ID.value: healthevent.storageset_id,
+            HealthAttr.NODE_ID.value: healthevent.node_id,
+            HealthAttr.RESOURCE_TYPE.value: healthevent.resource_type,
+            HealthAttr.RESOURCE_ID.value: healthevent.resource_id,
+            HealthAttr.RESOURCE_STATUS.value: healthevent.event_type,
```
Comment (Madhura-08) [inline ha/core/event_manager/model/action_event.py:40]: I think the last "," for line 40 is not required.

Comment (Madhura-08) [reply ha/core/event_manager/model/action_event.py:40]: Why the specific_info is removed?

Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py:40]: as for specific_info we created new function as it's nested dict, for set_payload function other attributes will set but for specific_info separately set with function set_specific_info() please check 43 line .

Comment (Madhura-08) [reply ha/core/event_manager/model/action_event.py:40]: Ok..But please address that "," comment for line number 40.

**Review by mariyappanp — APPROVED** [review]

File: ha/core/event_manager/model/action_event.py (hunk 7 above)
Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py]: Done

Comment (Madhura-08) [reply ha/core/event_manager/model/action_event.py]: Thanks.

**Review by Madhura-08 — APPROVED** [review]

**Comment by indrajitzagade** [comment]
@gauri-bhosale As discussed please implement the pending LLD comment: https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/891781277?focusedCommentId=916553923#comment-916553923

File: ha/core/event_manager/model/action_event.py [hunk 8]
```diff
@@ -29,20 +28,19 @@ def __init__(self, healthevent: HealthEvent):
...
+            HealthAttr.RACK_ID.value: healthevent.rack_id,
+            HealthAttr.STORAGESET_ID.value: healthevent.storageset_id,
+            HealthAttr.NODE_ID.value: healthevent.node_id,
+            HealthAttr.RESOURCE_TYPE.value: healthevent.resource_type,
+            HealthAttr.RESOURCE_ID.value: healthevent.resource_id,
+            HealthAttr.RESOURCE_STATUS.value: healthevent.event_type
         }
-        self.event.set_payload(payload)
+        self.event = HEvent(**payload)
+        self.event.set_specific_info({HealthAttr.SPECIFIC_INFO.value: healthevent.specific_info})
```
Comment (indrajitzagade) [inline ha/core/event_manager/model/action_event.py:43]: I guess we do not need to add the key: HealthAttr.SPECIFIC_INFO.value here. Please check.

File: ha/k8s_setup/ha_setup.py [hunk 9]
```diff
@@ -252,10 +252,10 @@ def process(self):
...
             self._cluster_id = Conf.get(self._index, f'node{_DELIM}{machine_id}{_DELIM}cluster_id')
             # site_id = Conf.get(self._index, f'node{_DELIM}{machine_id}{_DELIM}site_id')
-            self._site_id = '1'
+            self._site_id = NOT_DEFINED
             # rack_id = Conf.get(self._index, f'node{_DELIM}{machine_id}{_DELIM}rack_id')
-            self._rack_id = '1'
-            self._storageset_id = '1'
+            self._rack_id = NOT_DEFINED
+            self._storageset_id = NOT_DEFINED
             conf_file_dict.update({'COMMON_CONFIG': {'cluster_id': self._cluster_id, 'rack_id': self._rack_id, 'site_id': self._site_id}})
```
Comment (indrajitzagade) [inline ha/k8s_setup/ha_setup.py:259]: Can we push this storageset_id as well to the HA conf file? @ArchanaLimaye ?

**Comment by akash2144** [comment]
In health_monitor logs I can see there are different values for "storageset_id" & "generation_field" in [process_event] log_info & "Sending action event" log_info.
```
INFO [process_event] Evaluated {"source": "", "event_id": "1646653845.45331745eac0f31cfe84f31abbf107e050a74b4", "event_type": "online", "severity": "informational", "site_id": "1", "rack_id": "1", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "storageset_id": "89e0d99c15a549ddbdffbe018e2ab928", "node_id": "89e0d99c15a549ddbdffbe018e2ab928", "host_id": "89e0d99c15a549ddbdffbe018e2ab928", "resource_type": "node", "timestamp": "1646653845.4533174", "resource_id": "89e0d99c15a549ddbdffbe018e2ab928", "specific_info": {"generation_id": "cortx-data-ssc-vm-rhev4-2221-6f5dcf55f4-lqs6s"}} with action ['publish']
2022-03-07 11:50:45 health_monitor [8]: INFO [act] Default action handler with Event: 1646653845.45331745eac0f31cfe84f31abbf107e050a74b4 and actions : ['publish']
Sending action event {"header": {"version": "1.0", "timestamp": "1646678416.3081498", "event_id": "1646678416.308149871abae53283445ee88897f040793b3d0"}, "payload": {"source": "", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "site_id": "1", "rack_id": "1", "storageset_id": "eb4dd0e836454816a2b44961d3c9ef72", "node_id": "eb4dd0e836454816a2b44961d3c9ef72", "resource_type": "node", "resource_id": "eb4dd0e836454816a2b44961d3c9ef72", "resource_status": "online", "specific_info": {"specific_info": {"generation_id": null}}}} to component hare
```
Is this expected ?

File: ha/monitor/k8s/parser.py [hunk 10]
```diff
@@ -24,18 +24,18 @@
...
 class ObjectParser:
     def __init__(self):
-        self.payload = {HEALTH_ATTRIBUTES.SOURCE.value: HEALTH_EVENT_SOURCES.MONITOR.value, \
-                    HEALTH_ATTRIBUTES.CLUSTER_ID.value: None, HEALTH_ATTRIBUTES.SITE_ID.value: None, \
-                    HEALTH_ATTRIBUTES.RACK_ID.value: None, HEALTH_ATTRIBUTES.STORAGESET_ID.value: None, \
-                    HEALTH_ATTRIBUTES.NODE_ID.value: None, HEALTH_ATTRIBUTES.RESOURCE_TYPE.value: None, \
-                    HEALTH_ATTRIBUTES.RESOURCE_ID.value: None, HEALTH_ATTRIBUTES.RESOURCE_STATUS.value: None, \
-                    HEALTH_ATTRIBUTES.SPECIFIC_INFO.value: {}}
+        self.payload = {HealthAttr.SOURCE.value: HEALTH_EVENT_SOURCES.MONITOR.value,
+                    HealthAttr.CLUSTER_ID.value: None, HealthAttr.SITE_ID.value: None,
```
Comment (indrajitzagade) [inline ha/monitor/k8s/parser.py:34]: @gauri-bhosale Can we use the NOT_DEFINED constant here for site, rack and storageset_id?

File: ha/core/event_manager/model/action_event.py (hunk 8 above)
Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py:43]: Done

Comment (indrajitzagade) [reply ha/core/event_manager/model/action_event.py:43]: I see it is not removed yet @gauri-bhosale

File: ha/monitor/k8s/parser.py (hunk 10 above)
Comment (gauri-bhosale) [reply ha/monitor/k8s/parser.py]: Added NOT_DEFINED to site , rack and storageset_id

File: ha/core/event_manager/model/action_event.py (hunk 8 above)
Comment (gauri-bhosale) [reply ha/core/event_manager/model/action_event.py]: Fixed.

**Comment by gauri-bhosale** [comment]
> In health_monitor logs I can see there are different values for "storageset_id" & "generation_field" in [process_event] log_info & "Sending action event" log_info.
> 
> ```
> INFO [process_event] Evaluated {"source": "", "event_id": "1646653845.45331745eac0f31cfe84f31abbf107e050a74b4", "event_type": "online", "severity": "informational", "site_id": "1", "rack_id": "1", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "storageset_id": "89e0d99c15a549ddbdffbe018e2ab928", "node_id": "89e0d99c15a549ddbdffbe018e2ab928", "host_id": "89e0d99c15a549ddbdffbe018e2ab928", "resource_type": "node", "timestamp": "1646653845.4533174", "resource_id": "89e0d99c15a549ddbdffbe018e2ab928", "specific_info": {"generation_id": "cortx-data-ssc-vm-rhev4-2221-6f5dcf55f4-lqs6s"}} with action ['publish']
> 2022-03-07 11:50:45 health_monitor [8]: INFO [act] Default action handler with Event: 1646653845.45331745eac0f31cfe84f31abbf107e050a74b4 and actions : ['publish']
> Sending action event {"header": {"version": "1.0", "timestamp": "1646678416.3081498", "event_id": "1646678416.308149871abae53283445ee88897f040793b3d0"}, "payload": {"source": "", "cluster_id": "a14e59d5c9374714a2dc97523bf03e49", "site_id": "1", "rack_id": "1", "storageset_id": "eb4dd0e836454816a2b44961d3c9ef72", "node_id": "eb4dd0e836454816a2b44961d3c9ef72", "resource_type": "node", "resource_id": "eb4dd0e836454816a2b44961d3c9ef72", "resource_status": "online", "specific_info": {"specific_info": {"generation_id": null}}}} to component hare
> ```
> 
> Is this expected ?

copied log from other locations thats why it mismatched, Rehanged now retested node failure and added logs please check. @akash2144

**Review by akash2144 — APPROVED** [review]

File: ha/k8s_setup/ha_setup.py (hunk 9 above)
Comment (ArchanaLimaye) [reply ha/k8s_setup/ha_setup.py:259]: Yes we can do that

File: ha/k8s_setup/ha_setup.py [hunk 11]
```diff
@@ -252,10 +252,10 @@ def process(self):
             # So, to get the cluster_id field from Confstore, we need machine_id
             self._cluster_id = Conf.get(self._index, f'node{_DELIM}{machine_id}{_DELIM}cluster_id')
             # site_id = Conf.get(self._index, f'node{_DELIM}{machine_id}{_DELIM}site_id')
-            self._site_id = '1'
+            self._site_id = NOT_DEFINED
```
Comment (ArchanaLimaye) [inline ha/k8s_setup/ha_setup.py:255]: Can these be defined as constants . They need to be avilable to other components as well for initializing events.
ex: in fault_tolerance/const.py ?

File: ha/k8s_setup/ha_setup.py (hunk 9 above)
Comment (indrajitzagade) [reply ha/k8s_setup/ha_setup.py:259]: Ok, lets take it separately.

File: ha/k8s_setup/ha_setup.py (hunk 11 above)
Comment (indrajitzagade) [reply ha/k8s_setup/ha_setup.py:255]: @ArchanaLimaye Already it is defined in fault_tolerance/const.py file.

**Review by indrajitzagade — APPROVED** [review]

File: ha/fault_tolerance/const.py [hunk 12]
```diff
@@ -27,23 +27,4 @@ class FAULT_TOLERANCE_KEYS(enum.Enum, metaclass=EnumListMeta):
...
-    RESOURCE_TYPE = 'resource_type'
-    RESOURCE_ID = 'resource_id'
-    RESOURCE_STATUS = 'resource_status'
-    SPECIFIC_INFO = 'specific_info'
-
-# Class to define constants for health event attributes
-class EVENT_ATTRIBUTES(enum.Enum, metaclass=EnumListMeta):
-    HEALTH_EVENT_HEADER = 'header'
-    HEALTH_EVENT_PAYLOAD = 'payload'
+NOT_DEFINED = '0xBAADFOOD'
```
Comment (ArchanaLimaye) [inline ha/fault_tolerance/const.py:30]: Please add  comment here as to where this constant is used. Also add a TODO : To be replaced by actual values when they are available in config

**Comment by ArchanaLimaye** [comment]
@gauri-bhosale 

- In existing logs, Why is "source" filed empty in all the outputs ?
- storageset_id is not coming as NOT_DEFINED 

Can you please check 
- Also please  add  testing results for both monitor and Hare as source

**Comment by gauri-bhosale** [comment]
> @gauri-bhosale
> 
> * In existing logs, Why is "source" filed empty in all the outputs ?
> * storageset_id is not coming as NOT_DEFINED
> 
> Can you please check

Need to check this why source getting empty and storageset_id value, created separate ticket for that : 
https://jts.seagate.com/browse/CORTX-29282

**Review by ArchanaLimaye — APPROVED** [review]

**Review by shaileshsvc — APPROVED** [review]
Approved

**Pull request merged**